### ISSUE TITLE
Old nVidia, old driver, old CUDA, lots of warnings, card doesn't mine

### ISSUE THREAD

**Issue by Djago** [RainbowMiner/RainbowMiner#381]
I have an old nVidia GTX550Ti. nVidia page directs me to driver version 391.35 that has CUDA 9.1.84. The problem is that lots of miners need CUDA 9.2 or newer so I get lots of warnings telling me that:
WARNING: Bminer requires CUDA version 9.2 or above (installed version is 9.1.84). Please update your Nvidia drivers.
WARNING: CcminerAlexis78 requires CUDA version 10.0 or above (installed version is 9.1.84). Please update your Nvidia drivers.
WARNING: CcminerBalloon requires CUDA version 9.2 or above (installed version is 9.1.84). Please update your Nvidia drivers.
WARNING: CcminerCool requires CUDA version 10.0 or above (installed version is 9.1.84). Please update your Nvidia drivers.
WARNING: CcminerSupr requires CUDA version 9.2 or above (installed version is 9.1.84). Please update your Nvidia drivers.
WARNING: CcminerX16r requires CUDA version 9.2 or above (installed version is 9.1.84). Please update your Nvidia drivers.
WARNING: CcminerYescrypt requires CUDA version 10.0 or above (installed version is 9.1.84). Please update your Nvidia drivers.
WARNING: Excavator requires CUDA version 10.0 or above (installed version is 9.1.84). Please update your Nvidia drivers.
WARNING: Fireice requires CUDA version 10.0 or above (installed version is 9.1.84). Please update your Nvidia drivers.
WARNING: XmrigNvidia requires CUDA version 9.2 or above (installed version is 9.1.84). Please update your Nvidia drivers.
WARNING: Could not request from 127.0.0.1:31000
WARNING: Could not request from 127.0.0.1:31000
WARNING: Could not request from 127.0.0.1:31000
WARNING: Failed to connect to miner (Excavator1.4.4-1-GPU#00).

So my good ol' nVidia card can't mine with almost any miner. I can see that CryptoDredge (at least) can mine with current driver, but RM is not mining with it. I only see it mining with CPU and AMD (I have a VEGA56).
So, two things:
1) If miners can't work with current CUDA, wouldn't be nice if RM remove those miners automagically from the list of allowed miners for that card?
2) Why isn't my nVidia card mining with, say, CryptoDredge?

On a side question... Should I install a new driver? I know that nVidia dropped support for Fermi chips and new CUDA won't use new features, but won't a new driver be supported by new miners and *maybe* allow my card to mine? Or should I dig a hole in my yard and bury it?

**Comment by paulpoco** [comment]
Nvidia 550 will use more electricity than you will earn.  Not worth it.

**Comment by Djago** [comment]
Electricity here is very cheap (0.02u$/kW). Also I'm betting that the depressed cryptomarket will rise again so what I mine today will have a better value tomorrow.

**Comment by Djago** [comment]
I've tried installing a newer version of driver but I get a message telling me that there's no compatible device... I'll try installing CUDA Toolkit.

**Comment by Djago** [comment]
Didn't work. CUDA 9.2 is installed but RM still detects 9.1.

**Comment by Djago** [comment]
I've tried excluding the incompatible miners in the config for that card, but the messages still appear

**Comment by Djago** [comment]
I've tried manually running CryptoDredge (one of the only two "capable" miners detected by RM). I have the line:
CryptoDredge.exe -r 10 -R 1 -b 127.0.0.1:31300 -d 0 -a neoscrypt --no-watchdog -o stratum+tcp://hub.miningpoolhub.com:17012 -u "xxxxxx" -p x (with "xxxxxx" my data).
But I get:
[21:57:25] WARN  - The required computer capability version is 5 or later (current is: 2.1)
[21:57:25] WARN  - Asus GTX 550 Ti is not supported
[21:57:25] ERROR - There are no supported devices
So it seems RM incorrectly detected CryptoDredge as compatible?
Same thing happens with Excavator:
[22:32:40][0x00000860][info] Log started
[22:32:41][0x00000860][error] core | No compatible devices found!

To sum up: there seems to be no compatible miner with GTX550Ti and RM thought that Excavator and CryptoEdge are compatible.

**Comment by RainbowMiner** [comment]
Are any of the miners using the old nVidia? Actually, I think the fermi chipset has a very low coverage. Plus the nVidia drivers dropped support at CUDA 9.2 for it. Most of the integrated miners do need at least CUDA 9.2.
Anyway, I will change the warning messages, so that they do not pop up every round.

**Comment by RainbowMiner** [comment]
> WARNING: Could not request from 127.0.0.1:31600

This indicates a crashed (or stopped) miner

**Comment by RainbowMiner** [comment]
> CryptoDredge.exe -r 10 -R 1 -b 127.0.0.1:31300 -d 0 -a neoscrypt --no-watchdog -o stratum+tcp://hub.miningpoolhub.com:17012 -u "xxxxxx" -p x (with "xxxxxx" my data).
But I get:
[21:57:25] WARN - The required computer capability version is 5 or later (current is: 2.1)
[21:57:25] WARN - Asus GTX 550 Ti is not supported
[21:57:25] ERROR - There are no supported devices

CryptoDredge comes with a CUDA 9.1 version, this is why RainbowMiner tries to use it. Unfortunatly, it seem like the CryptoDredge kernel does not support the Fermi chipset. RBM checks for CUDA versions, only.

**Comment by Djago** [comment]
There's something wrong with this:
GeForce GTX 550 Ti (GPU#00)
===========================

Miner          Fee Algorithm       Speed Power USD/Day  XMR/Day  Accuracy Pool              PoolFee
-----          --- ---------       ----- ----- -------- -------- -------- ----              -------
CcminerDumax     - PHI2        1,79 GH/s    0W  70,2818 1,634342     94 % NLpool              0,1 %
CcminerTpruvot   - Allium    153,69 MH/s    0W   6,5532 0,152390     96 % Zpool              1,25 %
CcminerTpruvot   - Lyra2z    202,05 MH/s    0W   4,2050 0,097784     94 % BlazePool             1 %
CcminerSkunk     - Skunk       1,67 MH/s    0W   0,0045 0,000105     96 % Zpool-MobitGlobal  1,25 %

There's no way that my puny GTX550Ti generates 70USD a day!!!! 🤣 
6, and 4 isn't reasonable either 🤣 
Now 0.000105 is more like it 😭

**Comment by RainbowMiner** [comment]
Wow! What a beast, it is 🤣
You should disable PHI2 on this GPU. It looks like, it miscalculates this algo.
Edit Config\devices.config.txt, find the GTX550Ti section and put PHI2 into "ExcludeAlgorithm" field.
```
"ExcludeAlgorithm": "PHI2",
```

**Comment by Djago** [comment]
Done
What about Lyra2z and Allium?

**Comment by RainbowMiner** [comment]
Ya, for sure. Just delete the file "Stats\Profit.txt" and the historical numbers will be forgotten.
